noir-projects/aztec-nr/aztec/src/test/helpers/context_builder.nr

```diff
@@ -0,0 +1,52 @@
+use crate::context::{PrivateContext, PublicContext};
+use dep::protocol_types::address::AztecAddress;
+use dep::std::test::OracleMock;
+
+struct ContextBuilder {
+    block_number: Option<Field>,
+    contract_address: Option<AztecAddress>,
+}
+
+impl ContextBuilder {
+    fn new() -> Self {
+        Self { block_number: Option::none(), contract_address: Option::none() }
+    }
+
+    fn block_number(&mut self, block_number: Field) -> &mut Self {
+        self.block_number = Option::some(block_number);
+        self
+    }
+
+    fn contract_address(&mut self, contract_address: AztecAddress) -> &mut Self {
+        self.contract_address = Option::some(contract_address);
+        self
+    }
+
+    fn private(&mut self) -> PrivateContext {
+        let mut context = PrivateContext::empty();
+
+        if self.block_number.is_some() {
+            context.inputs.historical_header.global_variables.block_number = self.block_number.unwrap_unchecked();
+        }
+
+        if self.contract_address.is_some() {
+            context.inputs.call_context.storage_contract_address = self.contract_address.unwrap_unchecked();
+        }
+
+        context
+    }
+
+    fn public(&mut self) -> PublicContext {
+        let mut context = PublicContext::empty();
+
+        if self.block_number.is_some() {
+            let _ = OracleMock::mock("avmOpcodeBlockNumber").returns(self.block_number.unwrap());
```

Comment:
You know this, but observe that this is global so if you create 2 public contexts, I guess the last one mocked will prevail. It's kind of flakey (but there's no other way to do it AFAICT).

I had to do this in the AVM migration because the former test was modifying the context directly (which IDK if it's the idea in general)

Reply:
Hm true, but I don't think we'd ever create two contexts - definitely not two _public_ contexts, that'd make no sense in a test environment.